### PR TITLE
[WIP] Upgrade command for cfg

### PR DESCRIPTION
**Pull request by phanimarupaka**
@mortent 

This PR is to introduce upgrade command to upgrade setters from V1 to the latest.

### CODE REVIEW

File: kyaml/setters/lookupkio.go [hunk 1]
```diff
@@ -10,15 +10,18 @@ import (
...
+// LookupSettersDeprecated identifies setters for a collection of Resources
+type LookupSettersDeprecated struct {
 	// Name is the name of the setter to match.  Optional.
 	Name string
 
 	// SetterCounts is populated by Filter and contains the count of fields matching each setter.
 	SetterCounts []setterCount
+
+	// Upgrade the setters
+	Upgrade bool
```
Comment (mortent) [inline kyaml/setters/lookupkio.go:24]: Having a `LookupSetters` filter that mutates the setters seems wrong.

File: cmd/config/internal/commands/cmdupgrade.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		}.Execute()
+		if err != nil {
+			fmt.Fprintf(c.OutOrStdout(), "encountered error: %s while migrating setter %s\n", err.Error(), setter.Name)
+		} else {
+			upgradedCount++
+		}
+	}
+	fmt.Fprint(c.OutOrStdout(), "\n"+`if your resources contain partial setters, they are not automatically converted to substitutions,
+however the references are removed from resources and constituent setters are created,
+please use create-subst command to create equivalent substitutions using upgraded setters`+"\n\n")
```
Comment (mortent) [inline cmd/config/internal/commands/cmdupgrade.go:95]: We should try to do better here. Are we sure there are no way to automatically create substitutions from partial setters? If there isn't, can we be more helpful here in guiding the users. For example, print out the commands needed for creating substitutions that corresponds to the partial setters, but leave placeholders for information the user needs to fill in.

File: kyaml/setters/lookupkio.go [hunk 3]
```diff
@@ -10,15 +10,18 @@ import (
 	"sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
-var _ kio.Filter = &LookupSetters{}
+var _ kio.Filter = &LookupSettersDeprecated{}
 
-// LookupSetters identifies setters for a collection of Resources
-type LookupSetters struct {
+// LookupSettersDeprecated identifies setters for a collection of Resources
+type LookupSettersDeprecated struct {
```
Comment (mortent) [inline kyaml/setters/lookupkio.go:16]: If we are deprecating this, we need to add some more explanation for why and what is the preferred solution.

File: cmd/config/docs/commands/upgrade.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+## set
```
Comment (monopole) [inline cmd/config/docs/commands/upgrade.md:1]: set?

File: cmd/config/docs/commands/upgrade.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+Upgrade setters for Resources.
+
+### Synopsis
+
+Upgrade setters for Resources.
```
Comment (monopole) [inline cmd/config/docs/commands/upgrade.md:7]: You might want to make this help less specific, i.e. not call out _setters_ specifically.

kustomize has an `edit fix` command that reads in a kustomization file, purports to looks for deprecated stuff, converts it to whatever the new format should be, and writes it out.  It does more than just deal with deprecations - fields will be sorted, indented, etc. (like `go fmt`).

**Pull request closed without merging**